setup.py

```diff
@@ -8,6 +8,7 @@
 from distutils.cmd import Command
 import os.path
 import sys
+import warnings
```

Comment:
This file, one of the 3 largest in the project, increased in size to 362 lines.  The total size of those files is now 3404 lines (target: 1132).  Is this file complex enough to refactor?